pkg/plugins/user.go

```diff
@@ -69,30 +69,15 @@ func createUser(fs vfs.FS, u schema.User, console Console) error {
 	}
 
 	primaryGroup := u.Name
-	gid := 1000
 
+	gid := -1 // -1 instructs entities to find the next free id and assign it
 	if u.PrimaryGroup != "" {
 		gr, err := osuser.LookupGroup(u.PrimaryGroup)
 		if err != nil {
 			return fmt.Errorf("could not resolve primary group of user: %s", err.Error())
 		}
 		gid, _ = strconv.Atoi(gr.Gid)
 		primaryGroup = u.PrimaryGroup
-	} else {
-		// Create a new group after the user name
-		all, _ := entities.ParseGroup(etcgroup)
-		if len(all) != 0 {
-			usedGids := []int{}
-			for _, entry := range all {
-				usedGids = append(usedGids, *entry.Gid)
-			}
-			sort.Ints(usedGids)
-			if len(usedGids) == 0 {
-				return fmt.Errorf("no new guid found for group: %s", etcgroup)
-			}
-			gid = usedGids[len(usedGids)-1]
-			gid++
-		}
 	}
```

Comment:
Where did the `else` part go ?

Reply:
My understanding is that this part of the else was only looking for an available `gid`. With `git := -1` the library entities is instructed to actually find a suitable `gid`. I guess that's why there is a `gid = *group.Gid` on line 99 below. To actually retrieve the generated `gid` that entities library provided.

In other words, the generation of a suitable `gid` has been moved from actual yip's code to the entities library.